test/document/document_test.ts

```diff
@@ -346,6 +346,126 @@ describe('Document', function () {
     assert.equal(root, clone);
   });
 
+  it('garbage collection test for text', function () {
+    const doc = Document.create('test-col', 'test-doc');
+    assert.equal('{}', doc.toSortedJSON());
+
+    let expected_msg = '{"k1":"Hello mario"}';
+    doc.update((root) => {
+      const text = root.createText('k1');
+      text.edit(0, 0, 'Hello world');
+      text.edit(6, 11, 'mario');
+      assert.equal(expected_msg, root.toJSON());
+    }, 'edit text k1');
+    assert.equal(expected_msg, doc.toSortedJSON());
+    assert.equal(1, doc.getGarbageLen());
+
+    expected_msg = '{"k1":"Hi jane"}';
+
+    doc.update((root) => {
+      const text = root['k1'];
+      text.edit(0, 5, 'Hi');
+      text.edit(3, 4, 'j');
+      text.edit(4, 8, 'ane');
+      assert.equal(expected_msg, root.toJSON());
+    }, 'deletes 2');
+    assert.equal(expected_msg, doc.toSortedJSON());
+
+    const expectedGarbageLen = 4;
+    assert.equal(expectedGarbageLen, doc.getGarbageLen());
+    assert.equal(expectedGarbageLen, doc.garbageCollect(MaxTimeTicket));
+  });
+
+  it('garbage collection test for rich text', function () {
+    const doc = Document.create('test-col', 'test-doc');
+    assert.equal('{}', doc.toSortedJSON());
+
+    let expected_msg =
+      '{"k1":[{"attrs":{"b":"1"},"content":Hello },{"attrs":{},"content":mario},{"attrs":{},"content":\n}]}';
+    doc.update((root) => {
+      const text = root.createRichText('k1');
+      text.edit(0, 0, 'Hello world', { b: '1' });
+      text.edit(6, 11, 'mario');
+      assert.equal(expected_msg, root.toJSON());
+    }, 'edit rich text k1');
+    assert.equal(expected_msg, doc.toSortedJSON());
+    assert.equal(1, doc.getGarbageLen());
+
+    expected_msg =
+      '{"k1":[{"attrs":{"b":"1"},"content":Hi},{"attrs":{"b":"1"},"content": },{"attrs":{},"content":j},{"attrs":{"b":"1"},"content":ane},{"attrs":{},"content":\n}]}';
+
+    doc.update((root) => {
+      const text = root['k1'];
+      text.edit(0, 5, 'Hi', { b: '1' });
+      text.edit(3, 4, 'j', null);
+      text.edit(4, 8, 'ane', { b: '1' });
+      assert.equal(expected_msg, root.toJSON());
+    }, 'edit rich text k1');
+    assert.equal(expected_msg, doc.toSortedJSON());
+
+    const expectedGarbageLen = 4;
+    assert.equal(expectedGarbageLen, doc.getGarbageLen());
+    assert.equal(expectedGarbageLen, doc.garbageCollect(MaxTimeTicket));
```

Comment:
In my opinion, what can check the success of `GarbageCollect` is to check the number of garbage after collection.
So, it would be nice to add a check the number of garbage after collection at the end.

```suggestion
    assert.equal(expectedGarbageLen, doc.getGarbageLen());
    assert.equal(expectedGarbageLen, doc.garbageCollect(MaxTimeTicket));
    
    const empty = 0;
    assert.equal(empty, doc.getGarbageLen());
```

With this modification, this test can explain the final goal of `GarbageCollect`.